### DISCUSSION TITLE
关于复现论文的相关问题

### DISCUSSION THREAD

**Discussion by WangTingZheng** [WangTingZheng/Paperdb#17] [Show and tell]
## 复现论文的目标是什么？
- 尽可能地完美复现，也就是说不仅要把功能实现了，还得使得系统的其它部分仍然可以正常工作 
- 不能删除LevelDB的原有设计，除非论文里提及了
- 严格的检查，必须通过单元测试、Google Google sanitizers、跨平台多编译器的CI以及benchmark，必要时，还需要有专业人士的review
- 从README到文档，再到代码注释，从简单到细节的文档系统，保证其他人可以看得懂

## 复现论文不追求的是什么？
- 大架构的调整，比如说DiffKV
- 重新造一个LSM Tree的轮子

## 复现论文的难点在哪？
- 理解论文的思路
- 实现出论文里没有提到的细节，比如说ElasticBF中，哈希函数的种子对假阳率的影响
- 修复增加新功能对原系统的破坏

## 复现的论文的选择有什么讲究？
- 从外到里，分为三个层次
    - SSTable、Cache、MemTable
    - Compaction
    - MVCC/WAL
    - Write Stall
    - 其它小优化
 - 先把模块的代码还原为论文，再复现

**Comment by lokax** [comment]
:thinking:

**Comment by KKould** [comment]
:thinking:
